### ISSUE TITLE
DynComp instrumentation: Too many parameters for an annotation method

### ISSUE THREAD

**Issue by mernst** [codespecs/daikon#129]
I suspect this is a problem with DynComp instrumentation.

First, make sure that $DAIKONDIR is set.
```export DAIKONDIR=...```

Download needed libraries:
```
wget https://github.com/junit-team/junit4/releases/download/r4.12/junit-4.12.jar
wget http://search.maven.org/remotecontent?filepath=org/hamcrest/hamcrest-core/1.3/hamcrest-core-1.3.jar -O hamcrest-core-1.3.jar
```

Running JUnit works:
```
java -cp .:$DAIKONDIR/daikon.jar:junit-4.12.jar:hamcrest-core-1.3.jar org.junit.runner.JUnitCore
```

However, the identical command, just adding `daikon.DynComp`, fails:
```
java -cp .:$DAIKONDIR/daikon.jar:junit-4.12.jar:hamcrest-core-1.3.jar daikon.DynComp org.junit.runner.JUnitCore
```

Here is the output:
```
JUnit version 4.12
Exception in thread "main" java.lang.AssertionError: Too many parameters for an annotation method
	at sun.reflect.annotation.AnnotationInvocationHandler.invoke(AnnotationInvocationHandler.java:66)
	at com.sun.proxy.$Proxy0.annotationType(Unknown Source)
	at sun.reflect.annotation.AnnotationParser.parseAnnotations2(AnnotationParser.java:122)
	at sun.reflect.annotation.AnnotationParser.parseSelectAnnotations(AnnotationParser.java:101)
	at sun.reflect.annotation.AnnotationType.<init>(AnnotationType.java:145)
	at sun.reflect.annotation.AnnotationType.getInstance(AnnotationType.java:85)
	at sun.reflect.annotation.AnnotationParser.parseAnnotation2(AnnotationParser.java:266)
	at sun.reflect.annotation.AnnotationParser.parseAnnotations2(AnnotationParser.java:120)
	at sun.reflect.annotation.AnnotationParser.parseAnnotations(AnnotationParser.java:72)
	at java.lang.Class.createAnnotationData(Class.java:3521)
	at java.lang.Class.annotationData(Class.java:3510)
	at java.lang.Class.getAnnotation(Class.java:3415)
	at java.lang.reflect.AnnotatedElement.isAnnotationPresent(AnnotatedElement.java:258)
	at java.lang.Class.isAnnotationPresent(Class.java:3425)
	at org.junit.runner.notification.RunNotifier.wrapIfNotThreadSafe(RunNotifier.java:50)
	at org.junit.runner.notification.RunNotifier.addFirstListener(RunNotifier.java:212)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:134)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
	at org.junit.runner.JUnitCore.runMain(JUnitCore.java:77)
	at org.junit.runner.JUnitCore.main(JUnitCore.java:36)
```

**Comment by markro49** [comment]
Invocation of annotationType (via reflection) cannot have the dcomp_marker argument added.  Testing a fix now.

**Comment by markro49** [comment]
fixed by 639c429dea2bd39ef788aabdfe874deed89ca4b6
